src/or/connection.c

```diff
@@ -2727,9 +2768,40 @@ retry_all_listeners(smartlist_t *replaced_conns,
   if (retry_listener_ports(listeners,
                            get_configured_ports(),
                            new_conns,
+                           replacements,
                            close_all_noncontrol) < 0)
     retval = -1;
 
+  SMARTLIST_FOREACH_BEGIN(replacements, struct replacement_s *, r) {
+    int addr_in_use = 0;
+    int skip = 0;
+
+    tor_assert(r->new_port);
+    tor_assert(r->old_conn);
+
+    connection_t *new_conn =
+     connection_listener_new_for_port(r->new_port, &skip, &addr_in_use);
+
+    if (skip)
+      continue;
+
+    // XXX: replaced_conns
```

Comment:
What does this mean?

Reply:
We don't really have a way to walk back to old listener configuration in case this rebinding business fails. Then again, we're not implementing any walking back in general case either (when rebinding is not needed) in `options_act_reversible`. I guess this is fine, as long as we're not getting stuck in some bad state where we cannot just set the old torrc back?

Reply:
Should I delete this line?